addon/-private/system/identity-map.js

```diff
@@ -0,0 +1,28 @@
+import RecordMap from './record-map';
+import EmptyObject from './empty-object';
+
+export default class IdentityMap {
+  constructor() {
+    this._map = new EmptyObject();
```

Comment:
I switched to `Object.create(null)` but we should probably investigate which is currently better.